highlander/highlander.py

```diff
@@ -1,43 +1,59 @@
+from errno import EEXIST
 from logging import getLogger
-from os import getcwd, unlink
+from os import getcwd, mkdir
 from os.path import join, realpath, isfile
+from shutil import rmtree
 
 from psutil import Process, NoSuchProcess
 from funcy import decorator
 
-from .exceptions import InvalidPidFileError, PidFileExistsError
+from .exceptions import InvalidPidFileError, PidFileExistsError, InvalidPidDirectoryError
 
 logger = getLogger(__name__)
 default_location = realpath(join(getcwd(), '.pid'))
 
 
 @decorator
-def one(call, pid_file=default_location):
+def one(call, pid_directory=default_location):
     """ Check if the call is already running. If so, bail out. If not, create a
     file that contains the process identifier (PID) and creation time. After call
     has completed, remove the process information file.
     :param call: The original function using the @one decorator.
-    :param pid_file: The name of the file where the process information will be written.
+    :param pid_directory: The name of the directory where the process information will be written.
     :return: The output of the original function.
     """
-    if _is_running(pid_file):
+    if _is_running(pid_directory):
         logger.info('The process is already running.')
         return
 
-    _set_running(pid_file)
+    _set_running(pid_directory)
     try:
         result = call()
     finally:
-        _delete(pid_file)
+        _delete(pid_directory)
     return result
 
 
-def _is_running(pid_file):
+def _is_running(directory):
     """ Determine whether or not the process is currently running.
-    :param pid_file: The PID file containing the process information.
+    :param directory: The PID directory containing the process information.
     :return: Whether or not the process is currently running.
     """
-    if not isfile(str(pid_file)):
+    directory_exists = False
+    try:
+        mkdir(str(directory))
+    except OSError as e:
+        if e.errno == EEXIST:
+            directory_exists = True
+        else:
+            raise e
+
+    if not directory_exists:
+        return False
+
+    pid_file = _get_pid_filename(directory)
+
+    if not isfile(pid_file):
         return False
 
     pid, create_time = _read_pid_file(pid_file)
```

Comment:
So the use case you're talking about is what if the directory is deleted and it goes to read the PID file, but cannot do so because it was deleted? I suppose what I could do in this case is the following:

1. Attempt to open the file.
2. If I get an OSError from the file not existing, check if the directory exists.
3. If the directory does not exist, I can assume that the other process has cleaned it up and I should continue running.
4. If not, I should throw an IOError or ValueError like I currently do.

What do you think about that approach?